docs/Getting Started/Arch Linux/Arch Linux Root on ZFS.rst

```diff
@@ -508,13 +508,12 @@ System Installation
         zfs create -o canmount=off rpool_$INST_UUID/sys/DATA/default/$i
     done
 
-    for i in {home,root,srv,usr/local,var/log,var/spool,var/tmp};
```

Comment:
I don't think this is a good idea. /var/log and /var/spool should be separate from the system dataset.

If I roll back my OS dataset, I should not lose user data. /var/spool contains user data like email (if traditional /var/spool/mail is used) and cron jobs. I realize that /var/spool may not be used much on modern, single-user systems, but when it is used, it definitely contains user data.

/var/log is definitely used and definitely contains user data. If I rollback my OS, I do not want to roll back logs. I might need those logs to understand what broke on the system, leading to the rollback. I might also need those logs on a server for security/regulatory reasons.

/var/tmp can go either way. It's not used all that much, so it doesn't matter too much either way. An argument for separating it out is that you can skip snapshotting it. It looks like the current Ubuntu HOWTO no longer separates out /var/tmp. I feel like I used to.

Reply:
With boot environments, OS dataset is never rolled back. If going back to a previous state is needed, just reboot and select the clone. 

If the user needs to investigate the reason why the system is broken, she should mount the dataset then chroot into it. In this case, `/var/log` should preserve the log up to the broken point and should not be written by other boot environments.. If `/var/log` is shared, after chroot, a bunch of newly written messages will pop up in logs, making debugging a little more confusing.

The point is, system state should be preserved & sealed in a clone whenever possible, instead of mixing system states. 

I agree with you regarding `/var/spool`, added back.

Reply:
I agree with the statement about system states. I do not consider /var/log to be system state, nor did the Ubuntu guys at Canonical in their extensive conversations on what is and is not user data. (And they had a bunch of pre-existing experience from making similar distinctions on the now-defunct Ubuntu phone platform.) I very much think that /var/log should be treated as user data.

Pretend I used your approach on my email server. I apply updates. A while later, I notice that something is wrong. I troubleshoot, but ultimately decide I need to go back, at least temporarily, to get things working for customers again. I boot back into the previous boot environment clone (which I'm still calling a "roll back" even though it's not a literal `zfs rollback`). Now everything is working. Then a customer asks me about a message that did not come through. (This is not at all unusual under normal circumstances, but given that _we just had problems_ it's quite likely to occur.) Now I have a big hole in my logs? That's not okay. And sure, I can manually mount the clone, but that's extra work for no good reason.

For another example, substitute a DHCP server. I need a record of all leases for some period of time, so I can deal with abuse complaints. Your approach would again require me to mount the old BE to get access to the old /var/log. That also means I need to keep the old BE for however long my log retention policy requires. But I might not normally want to keep old BEs that long. This becomes a mess very quickly, for practical scenarios that I deal with routinely.

/var/log is user data and needs to be treated as such.

Reply:
I agree. Added back.